gui/canvaspainter/v7/src/TCanvasPainter.cxx

```diff
@@ -569,8 +568,8 @@ void ROOT::Experimental::TCanvasPainter::ProcessData(unsigned connid, const std:
       cdata.erase(0, 8);
       conn->fGetMenu = cdata;
    } else if (arg == "QUIT") {
-      if (gApplication)
-         gApplication->Terminate(0);
+      // use window manager to correctly terminate http server
+      TWebWindowsManager::Instance()->Terminate();
```

Comment:
In a future PR, could you convert `TWebWindowsManager` to have static functions, instead of having to go through `Instance()->` all the time? These static functions can then call `TWebWindowsManagerImpl::Instance()->` but at least the interface is simpler. Would that be an option?

Reply:
Generally speaking `TWebWindowsManager` is wrapper around `THttpServer` functionality.
And in multi-threaded environment I can imagine to have two (or more) independent instances of THttpServer for running different kinds of GUI elements. Probably, later we will provide interface to specify which window manager is used to create Canvas window(s)
Therefore for the moment I would keep interface as is - through `Instance()` and revise this functionality at the end.